### PR TITLE
Add tsan build on RBE and update image SHA.

### PR DESCRIPTION
**Pull request by adelez**

### CODE REVIEW

File: tools/internal_ci/linux/grpc_tsan_on_foundry.sh [hunk 1]
```diff
@@ -0,0 +1,87 @@
+Skip to content
```
Comment (matt-kwong) [inline tools/internal_ci/linux/grpc_tsan_on_foundry.sh]: Assuming lines 1-18 are accidentally included :)

File: tools/internal_ci/linux/grpc_tsan_on_foundry.sh [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  --strategy=Closure=remote  \
+  --genrule_strategy=remote  \
+  --experimental_strict_action_env=true \
+  --experimental_remote_platform_override='properties:{name:"container-image" value:"docker://gcr.io/asci-toolchain/nosla-debian8-clang-fl@sha256:496193842f61c9494be68bd624e47c74d706cabf19a693c4653ffe96a97e43e3" }' \
+  --crosstool_top=@com_github_bazelbuild_bazeltoolchains//configs/debian8_clang/0.2.0/bazel_0.7.0:toolchain \
+  --define GRPC_PORT_ISOLATED_RUNTIME=1 \
+  --copt=-fsanitize=thread \
+  --linkopt=-fsanitize=thread \
+  -- //test/...
+© 2018 GitHub, Inc.
```
Comment (matt-kwong) [inline tools/internal_ci/linux/grpc_tsan_on_foundry.sh]: These are also accidentally included?

File: tools/internal_ci/linux/grpc_tsan_on_foundry.sh [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -ex
+
+# A temporary solution to give Kokoro credentials.
```
Comment (matt-kwong) [inline tools/internal_ci/linux/grpc_tsan_on_foundry.sh]: Can this be changed to a TODO if we know what the future solution might be?

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

File: tools/internal_ci/linux/grpc_tsan_on_foundry.sh (hunk 1 above)
Comment (adelez) [reply tools/internal_ci/linux/grpc_tsan_on_foundry.sh]: Yeah. Fixed. :)

File: tools/internal_ci/linux/grpc_tsan_on_foundry.sh (hunk 3 above)
Comment (adelez) [reply tools/internal_ci/linux/grpc_tsan_on_foundry.sh]: Done

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

File: tools/internal_ci/linux/grpc_tsan_on_foundry.sh (hunk 1 above)
Comment (matt-kwong) [reply tools/internal_ci/linux/grpc_tsan_on_foundry.sh]: I see the extras at the end of file removed, but I still see the original lines 1-18

Comment (adelez) [reply tools/internal_ci/linux/grpc_tsan_on_foundry.sh]: Ah sorry. Done.

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Pull request merged**